src/error.rs

```diff
@@ -0,0 +1,14 @@
+use std::str;
```

Comment:
I really don't like having all the errors in a separate file for some reason. I think error structs should live in the modules where the error comes from. Since this is an opinionated view I'll wait for @est31's review

Reply:
In lewton I have a hybrid approach: there are separate error types for header decoding and audio decoding. Both of them live close to the code the error comes from. To offer an unified error to the outside, I've put an enum containing the two errors into lib.rs. Maybe I should have put it into a separate error.rs file. Does this sound like a good idea?

Reply:
Sounds good!